### ISSUE TITLE
False positive with isort rule I001 in jupyter notebooks

### ISSUE THREAD

**Issue by felix-cw** [astral-sh/ruff#5945]
Short summary: I think this problem is very similar to https://github.com/nbQA-dev/nbQA/issues/806

I am using the current latest version: `0.0.278`
This problem arises when you have a jupyter notebook containing a cell with imports without a trailing newline, followed by another cell without a leading newline, and you run ruff with rule `I001` selected.

My example notebook has the following cells:

```python
import os
```

```python
print("hello world")
```

If I run the command below, it shows an I001 violation, because there is no blank line separating the imports from the rest of the code.

```bash
ruff check ruff-isort.ipynb --no-cache --select=I001 --isolated --show-source
```

```text
ruff-isort.ipynb:cell 1:1:1: I001 [*] Import block is un-sorted or un-formatted
  |
1 | / import os
2 | | print("hello world")
  | |_^ I001
  |
  = help: Organize imports

Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

Also, when I run the command with the --fix option, no changes are made to the file and the error persists.

As in the nbQA issue mentioned above, the issue might be addressed by adding 2 newlines between cells instead of 1 in the intermediate code.

Thanks all for this amazing tool!

**Comment by dhruvmanila** [comment]
Hey, thanks for opening this issue and using Ruff on Jupyter Notebooks during the experimental phase! :)

This was fixed in #5876 and will be available in the next release.

**Comment by charliermarsh** [comment]
Thank you @felix-cw, really appreciate the issue!
